### PR TITLE
Small refactor for loading PRs

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

File: models/issues/pull_list.go [hunk 1]
```diff
@@ -175,6 +175,7 @@ func (prs PullRequestList) loadAttributes(ctx context.Context) error {
 	}
 	for i := range prs {
 		prs[i].Issue = set[prs[i].IssueID]
+		prs[i].Issue.PullRequest = prs[i] // if panic here, means issueIDs and prs are not in sync
```
Comment (delvh) [inline models/issues/pull_list.go]:
```suggestion
	for _, pr := range prs {
		pr.Issue = set[pr.IssueID]
		pr.Issue.PullRequest = pr // panic here means issueIDs and prs are not in sync
```

**Comment by delvh** [comment]
Oh, also: There was a panic already in https://drone.gitea.io/go-gitea/gitea/66309/2/15 :822 for another PR, so there will probably still be a panic.

**Comment by lunny** [comment]
> Oh, also: There was a panic already in https://drone.gitea.io/go-gitea/gitea/66309/2/15 :822 for another PR, so there will probably still be a panic.

That means fixtures data are broken.

**Review by delvh — APPROVED** [review]

**Review by yardenshoham — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by jolheiser** [comment]
:trumpet: :robot:

**Pull request merged**